packs/autoscale/actions/workflows/asg_deflate.yaml

```diff
@@ -16,83 +16,82 @@ workflows:
       get_chatops_room:
         action: st2.kv.get
         input:
-          key: 'asg.{$.asg}.channel'
+          key: 'asg.<% $.asg %>.channel'
         publish:
-          channel: '{$.get_chatops_room.result}'
+          channel: '<% $.get_chatops_room.result %>'
         on-success:
           - get_current_epoch
       get_current_epoch:
         action: st2.epoch
         publish:
-          current_epoch: $.get_current_epoch.result
+          current_epoch: <% $.get_current_epoch.result %>
         on-success:
           - get_last_deflate_timestamp
       get_last_deflate_timestamp:
         action: st2.kv.get
         input:
-          key: 'asg.{$.asg}.last_deflate_timestamp'
+          key: 'asg.<% $.asg %>.last_deflate_timestamp'
         publish:
-          last_deflate_timestamp: int($.get_last_deflate_timestamp.result)
+          last_deflate_timestamp: <% int($.get_last_deflate_timestamp.result) %>
         on-success:
           - get_deflate_delay
       get_deflate_delay:
         action: st2.kv.get
         input:
-          key: 'asg.{$.asg}.deflate_delay'
+          key: 'asg.<% $.asg %>.deflate_delay'
         publish:
-          deflate_delay: int($.get_deflate_delay.result)
+          deflate_delay: <% int($.get_deflate_delay.result) %>
         on-success:
           - get_deflate_by
       get_deflate_by:
         action: st2.kv.get
         input:
-          key: 'asg.{$.asg}.deflate_by'
+          key: 'asg.<% $.asg %>.deflate_by'
         publish:
-          deflate_by: int($.get_deflate_by.result)
+          deflate_by: <% int($.get_deflate_by.result) %>
         on-success:
-          - chatops_delete_nodes: $.force or
-              (($.last_deflate_timestamp + ($.deflate_delay * 60)) < $.current_epoch)
+          - chatops_delete_nodes: <% $.force %> or
+              ((<% $.last_deflate_timestamp %> + (<% $.deflate_delay %> * 60)) < <% $.current_epoch %>)
         on-error:
           - chatops_too_soon
       chatops_delete_nodes:
         action: slack.post_message
         input:
-          message: "ASG {$.asg} is a little bloated, it seems. Going to trim by {$.deflate_by} nodes..."
-          channel: '{$.channel}'
+          message: "ASG is a little bloated, it seems. Going to trim by <% $.deflate_by %> nodes..."
+          channel: '<% $.channel %>'
```

Comment:
Channel is a user configurable value on a per-asg basis, and that's by design. Would be good to know if I can call config from workflows as a general rule though. 

-James

> On Feb 26, 2015, at 11:18 PM, Dmitri Zimin(e) <notifications@github.com> wrote:
> 
> In packs/autoscale/actions/workflows/asg_deflate.yaml:
> 
> >          on-error:
> >            - chatops_too_soon
> >        chatops_delete_nodes:
> >          action: slack.post_message
> >          input:
> > -          message: "ASG {$.asg} is a little bloated, it seems. Going to trim by {$.deflate_by} nodes..."
> > -          channel: '{$.channel}'
> > +          message: "ASG is a little bloated, it seems. Going to trim by <% $.deflate_by %> nodes..."
> > +          channel: '<% $.channel %>'
> 1) check if @Kami fixed slack to take channel from config
> 2) check if ' ' are needed, YAQL now supposed to preserve types.
> 
> —
> Reply to this email directly or view it on GitHub.
>